### DISCUSSION TITLE
Does anyone know what 5.2 is getting at?

### DISCUSSION THREAD

**Discussion by jdonwells** [TEALSK12/introduction-to-computer-science#366] [Q&A]
Part 2.2 says "What happens to the original ("master") sprite each time the spacebar is pressed? Does that seem useful? What role should the original sprite play now that we're cloning?"

What is this getting at? Sprites in Snap! like objects in Javascript use a prototype instead of a class. Is that what this is about?

The bonus says "Assign each clone a different value for some properties, such as speed, color, or size. Try controlling these values from the master sprite rather than having each clone choose its own." Why would I do that in a prototyping language? 

As far as I can tell the original sprite is special in that it does not show up in the **my clones** block. So if the original sprite (I don't think I would call it master sprite) uses **my clones** it shows all of the clones, but if a clone uses **my clones** it shows the clones minus itself and minus the original sprite.

**Comment by aspiece** [comment]
@jdonwells  this is great feedback, I would love an issue or pull request on how you think it should be worded. Would love to see how you resolved this so we could incorporate it.

Andrew

**Reply by jdonwells** [reply]
https://docs.google.com/presentation/d/15D0cZ7bBZC2n_GuStTyj7UsI89QQZZPUO7ehIs8KRZU/edit?usp=sharing
I left the lab alone except for changing master sprite to prototype sprite. I expanded the slide show to include multiple examples and explained why the stage is a useful place to put the clone block. The explanation is a bit of a giveaway, but they have already written all of that code except for the clone block anyway.

The biggest changes I made were actually in 5.1 and 5.3.

5.1 is a problem in the age of COVID-19. So I created a simple online exercise. https://docs.google.com/document/d/1hQMYcUUL4NGu2LnwxP6-NG-UHlBQpFMdtzzn22SRyEA/edit?usp=sharing A simple piece of code creates a complex drawing. Some of the kids have chosen to take this and expand on it as their final project. So they must have liked it. I created a new project to show them how to do it with recursion instead of clones and inspire them in the shapes they can create with a few parameters. https://snap.berkeley.edu/snap/snap.html#present:Username=jdonwells&ProjectName=project%206.1%20Recursion%20Art%20Generator

5.3 is about controlling the sprites by giving each an id. Snap! has better ways to do that. Perhaps the lesson was about for this sprite only variables. I am not sure, but it isn't something they need.
https://docs.google.com/presentation/d/1oImv2YoEwAOEPHHkEHMm8C76595NFzPwFU6FtkxeHQw/edit?usp=sharing
https://docs.google.com/document/d/1L2aRw5FfQ1wr5QQ0JJljXbswV8juwZRU3eH68egJ2is/edit?usp=sharing

The big idea for unit 5 could be simple behavior in a single object can be complex behavior in a group. In that case I would use the game of life as the final project instead of space invaders which is just another arcade game.
